### PR TITLE
[DOCS] Makes PUT inference API docs collapsible

### PR DESCRIPTION
**Pull request by szabosteve**
This PR collects the properties of the PUT inference trained model API into collapsible sections.

Preview: http://elasticsearch_54653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-inference.html

### CODE REVIEW

**Review by benwtrent — APPROVED** [review]

File: docs/reference/ml/df-analytics/apis/put-inference.asciidoc [hunk 1]
```diff
@@ -61,257 +61,193 @@ The {infer} definition for the model. If `definition` is specified, then
...
-include::{docdir}/ml/ml-shared.asciidoc[tag=inference-config-classification-top-classes-results-field]
-=====
-====
-//End of inference_config
-
-//Begin input
-`input`::
+//Begin frequency encoding
+`frequency_encoding`::
 (Required, object)
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/put-inference.asciidoc:74]: @benwtrent Are all of `frequency_encoding`, `one_hot_encoding`, and

File: docs/reference/ml/df-analytics/apis/put-inference.asciidoc [hunk 2]
```diff
@@ -324,35 +260,160 @@ consequently, the probability of class 0 is `1 - P_1`. The class with the
...
+`metadata`::
+(Optional, object) 
+An object map that contains metadata about the model.
+
+`tags`::
+(Optional, string) 
+An array of tags to organize the model.
+
 
 [[ml-put-inference-example]]
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/put-inference.asciidoc:418]: IMO it's hard to figure out how all these partial examples fit together.  Could we provide one full example and then just link to https://github.com/elastic/ml-json-schemas/tree/master/examples for other combinations?

File: docs/reference/ml/ml-shared.asciidoc [hunk 3]
```diff
@@ -1242,10 +1237,10 @@ importance] values per document. By default, it is zero and no feature importanc
 calculation occurs.
 end::inference-config-regression-num-top-feature-importance-values[]
 
-tag::inference-config-regression-results-field[]
-Specifies the field to which the inference prediction is written. Defaults to
-`predicted_value`.
-end::inference-config-regression-results-field[]
+tag::inference-config-results-field[]
+The field that is added to incoming documents to contain the inference
```
Comment (lcawl) [inline docs/reference/ml/ml-shared.asciidoc:1241]: FYI I merged the two results_field definitions per discussions related to https://github.com/elastic/elasticsearch/pull/54421

**Review by lcawl — APPROVED** [review]
LGTM too

**Pull request merged**